lib/percy/client/environment.rb

```diff
@@ -28,33 +28,48 @@ def self.current_ci
 
       # @return [Hash] All commit data from the current commit. Might be empty if commit data could
       # not be found.
+      # Try getting data from git itself.
+      # If git is not present, fallback on data from environment variables.
       def self.commit
-        output = _raw_commit_output(_commit_sha) if _commit_sha
-        output ||= _raw_commit_output('HEAD')
-        return {branch: branch} unless output && output != ''
-        output = output.force_encoding('UTF-8') if output && output.encoding.to_s == 'US-ASCII'
+        git_data_from_git = _raw_commit_output(_commit_sha || 'HEAD')
```

Comment:
This is not logically equivalent.  Lets be sure to retry the command with `HEAD` if it fails with the `_commit_sha`

Reply:
Done!